web_poet/pages.py

```diff
@@ -64,9 +71,74 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
         return await item_from_fields(
-            self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
+            self,
+            item_cls=self.item_cls,
+            skip_nonitem_fields=self._skip_nonitem_fields,
+            field_names=self.fields_to_extract,
         )
 
+    def _get_select_fields(self) -> Optional[SelectFields]:
```

Comment:
\`\_get\_select\_fields()\` is not being cached.

I think this should be okay since the user might want to re-use the same page instance but with different \`SelectFields\`:

\`\`\`python  
page = SomePage(SelectFields(fields={"\*": False, "x": True}))  
item1 = page.to\_item()

page.select\_fields = SelectFields(fields={"\*": False, "x": True, "y": True})  
item2 = page.to\_item()  
\`\`\`

Any thoughts about this usage?

Reply:
I am OK with not caching.

However, I think we should avoid covering this scenario in the docs, and I am not sure if we should even cover it in tests. Changing the value of `select_fields` does not feel like a very elegant API to me.